### ISSUE TITLE
AI switching can cause a Player mon to turn into a bad egg

### ISSUE THREAD

**Issue by Land0fChocolate** [rh-hideout/pokeemerald-expansion#2672]
A Pokemon in the Player's party may turn into a bad egg and it probably has to do with the AI switching.

I've managed to reproduce the bug with these steps:
1.Fight Cooltrainer Wendy 
2.Use Marshtomp to faint her first Pokemon
3.Switch out Marshtomp for Crobat right after she sends out her second Pokemon (turn 2). 
4.Look at party menu again. Pokemon in slot 4 is now a bad egg. Pokemon in slot 3 also has bugged HP.

https://user-images.githubusercontent.com/52083703/217938497-e60f5d47-fc58-4ae1-808f-87e7a35fda8f.mp4

This are the details for Trainer Wendy in `src/data/trainers.h`:
```
[TRAINER_WENDY] =
    {
        .partyFlags = F_TRAINER_PARTY_CUSTOM_MOVESET,
        .trainerClass = TRAINER_CLASS_COOLTRAINER,
        .encounterMusic_gender = F_TRAINER_FEMALE | TRAINER_ENCOUNTER_MUSIC_COOL,
        .trainerPic = TRAINER_PIC_COOLTRAINER_F,
        .trainerName = _("Wendy"),
        .items = {ITEM_NONE, ITEM_NONE, ITEM_NONE, ITEM_NONE},
        .doubleBattle = FALSE,
        .aiFlags = AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_TRY_TO_FAINT | AI_FLAG_SETUP_FIRST_TURN,
        .partySize = ARRAY_COUNT(sParty_Wendy),
        .party = {.NoItemCustomMoves = sParty_Wendy},
    },
```

-If Crobat is moved to other slots except 3, and you switch with Crobat then slot 4 Pokemon turns into a bad egg. If Crobat is moved to slot 3, no bad eggs are made.
-If switch with Crobat on first turn, no issues.
-If switch to any other Pokemon instead of Crobat, no issues.
-Adding 3 more Pokemon to Wendy’s party changes nothing.
-Changing up Wendy’s party (so that Roselia is in slot 5) changes nothing.
-Changing the Pokemon in front from Marshtomp to something else doesn’t result in a bad egg. The HP bug still happens though.

Reverting this PR https://github.com/rh-hideout/pokeemerald-expansion/pull/2226 seemed to fix the bad egg bug.

**Comment by DizzyEggg** [comment]
@Land0fChocolate Couldn't reproduce on master. Could you try to reproduce the issue on a clean upcoming or master branch?

**Comment by Land0fChocolate** [comment]
I can't reproduce this on master either. I cannot be sure if there is something incompatible on my end that is causing the bug.

If nobody else is getting bad eggs or ? HP bars then may as well close this.

**Comment by AaghatIsLive** [comment]
got a similar bug from the battle tent,

![Screenshot 2023-04-16 170318](https://user-images.githubusercontent.com/109757010/232307183-887d5608-a285-4917-bbf8-6216cdece4ce.png)

**Comment by AsparagusEduardo** [comment]
> got a similar bug from the battle tent,

What version are you using?

**Comment by AaghatIsLive** [comment]
> What version are you using? 

I am on upcoming, my last pull had  #2898 as the latest commit.

**Comment by Jaizu** [comment]
MGBA has an option to rewind, I'm pretty sure some emulators might have this too. Having that enabled might help to record this issue if it happens again in the future.
Also, afaik, this couldn't be reproduced on vanilla RHH, so taking this issue with a grant of salt.

**Comment by AsparagusEduardo** [comment]
Hmm, we need more info, like what Pokémon/Moves the opponent had. This is reminding me of the Beat Up issue we had a bit back.

**Comment by AaghatIsLive** [comment]
In the battle tent it happens even if you forfeit the match and it doesn't always happen. Now that you mention it could be a specific mon, I'll do some testing...

**Comment by Bassoonian** [comment]
Is this still reproduceable in 1.7?

**Comment by AlexOn1ine** [comment]
> Is this still reproduceable in 1.7?

Nobody knows xD

It's a very hard issue to reproduce.

**Comment by DizzyEggg** [comment]
I would close this issue once #3277 is fixed. And then re-open or submit a new if it still happens.

**Comment by Bassoonian** [comment]
(Presumably) fixed by #3784
